src/sentry/static/sentry/app/views/performance/traceDetails/utils.ts

```diff
@@ -19,20 +20,38 @@ export function getTraceDetailsUrl(
 }
 
 function traceVisitor() {
-  const projectIds = new Set();
-  const eventIds = new Set();
+  const errorProjectSlugs = new Set();
+  const errorIds = new Set();
+  const transactionProjectSlugs = new Set();
+  const transactionIds = new Set();
 
-  return (accumulator, event) => {
-    if (!projectIds.has(event.project_id)) {
-      projectIds.add(event.project_id);
-      accumulator.totalProjects += 1;
+  return (accumulator: TraceInfo, event: TraceFull) => {
+    for (const error of event.errors ?? []) {
+      if (!errorProjectSlugs.has(error.project_slug)) {
+        errorProjectSlugs.add(error.project_slug);
 
-      // No user conditions yet, so all projects are relevant.
-      accumulator.relevantProjects += 1;
+        // No user conditions yet, so all projects are relevant to the error.
+        accumulator.relevantProjectsWithErrors += 1;
+      }
+
+      if (!errorIds.has(error.event_id)) {
+        errorIds.add(event.event_id);
+        accumulator.totalErrors += 1;
```

Comment:
Do we need to do all this if its a set? can't we just use `size` on the `errorIds` set and just add to the set regardless instead of checking `has` every time?

Reply:
That is a good point, I can update it to use the set size at the end instead of incrementing this counter repeatedly.